static/js/reducers/coupons.js

```diff
@@ -43,6 +46,8 @@ export const coupons = (state: CouponsState = INITIAL_COUPONS_STATE, action: Act
     };
   case RECEIVE_FETCH_COUPONS_FAILURE:
     return { ...state, fetchGetStatus: FETCH_FAILURE};
+  case SET_ACTIVE_COUPON:
```

Comment:
Instead of using `activeCoupon` can we just refetch from the coupon list API? I think this will help with race conditions, although there's still this possibility:

    GET                         response
    POST    GET    response

Which would cause stale data. But I think it would be a good start to refetch from the coupon list API